src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,48 @@
+@import "./utils/_variables" ;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: 706px;
+
+  @each $name, $size in $days {
+    &--start-day-#{$name} &__day:first-child {
+      margin-left: $size;
+    }
+  }
+
+  @for $j from 28 through 31 {
+    &--month-length-#{$j} &__day:nth-child(n + #{$j + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: $day-size;
+    width: $day-size;
+    border: 1px solid #000;
+    background: #eee;
+    padding: 10px;
+    transition: all ease 0.5s;
```

Comment:
Don't use `all`. Instead, list all values for this transition. It will make it way easier to understand what is it for.